### PR TITLE
Added rest_total_hits_as_int test for _search

### PR DESCRIPTION
**Pull request by awick**
### Description
Added simple search endpoint with rest_total_hits_as_int tests

### Issues Resolved

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tests/_core/search/rest_total_hits_as_int.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+$schema: ../../../json_schemas/test_story.schema.yaml
+
+description: Test search endpoint with rest_total_hits_as_int
```
Comment (dblock) [inline tests/_core/search/rest_total_hits_as_int.yaml:3]: Add a period. I started adding a linter in https://github.com/opensearch-project/opensearch-api-specification/pull/463.

File: tests/_core/search/rest_total_hits_as_int.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        director: Nicolas Winding Refn
+        title: Drive
+        year: 2011
+    status: [201]
+epilogues:
+  - path: /movies
+    method: DELETE
+    status: [200, 404]
+chapters:
+  - synopsis: Search with rest_total_hits_as_int=true
```
Comment (dblock) [inline tests/_core/search/rest_total_hits_as_int.yaml:30]: Period.

File: CHANGELOG.md [hunk 3]
```diff
@@ -59,6 +59,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added a warning on mulitple paths being tested in the same file ([#452](https://github.com/opensearch-project/opensearch-api-specification/pull/452))
 - Added `/_plugins/_query/settings` ([#456](https://github.com/opensearch-project/opensearch-api-specification/pull/456))
 - Added `/_plugins/_ppl`, `explain` and `stats` ([#460](https://github.com/opensearch-project/opensearch-api-specification/pull/460))
+- Added `rest_total_hits_as_int` ([#466](https://github.com/opensearch-project/opensearch-api-specification/pull/466))
```
Comment (dblock) [inline CHANGELOG.md:62]: This is just tests, no spec changes, remove this.

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good. Needs a rebase and some periods :)

**Comment by dblock** [comment]
It thinks `Refn` is a word and tries to spell-check it, add to .cspell or rename the movie.

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@awick I have this intermittently failing in some environments, I think the results need a sort. For now I'll reduce this to 1 record in #483.

**Comment by awick** [comment]
@dblock ok, the point of the 2 was to make sure track_total_hits was being honored and passed thru, when I get a chance I'll add the sort.

**Pull request merged**